akd/src/client.rs

```diff
@@ -215,34 +204,35 @@ pub fn key_history_verify<H: Hasher>(
     }
 
     // Check that all the individual update proofs check
```

Comment:
Can we add a comment what the actual `check` here is? Or better yet move it to a separate function with an apropos name.

Reply:
Should we also create a test for it?

Reply:
We can remove these epochs for sure -- just need to access them and make sure we do all the same checks as we are doing with the listed epochs. Thanks so much for noticing this!

Reply:
Separate function should also help with the visibility of the checks that are being done here.

Reply:
There were two checks:
1. Checking that the listed epochs are the same length as the number of update proofs. This check is now obsolete because the epochs are within the update proofs.
2. Listed epochs should be decreasing in value - this check has been ported to act on update proofs.

So we should be good! I've updated this comment to be a bit clearer that this loop verifies single update proofs.

I've also added a simple test asserting failure if the update proofs are in the wrong order (decreasing versions and epochs).

Reply:
Awesome! Thank you @afterdusk! :) Feel free to merge after CI!